code/game/objects/items/storage/belt.dm

```diff
@@ -9,23 +9,39 @@
 	slot_flags = SLOT_BELT
 	attack_verb = list("whipped", "lashed", "disciplined")
 	max_integrity = 300
-	var/content_overlays = FALSE //If this is true, the belt will gain overlays based on what it's holding
+	var/list/content_overlays = list() //If this is true, the belt will gain overlays based on what it's holding
 
 /obj/item/storage/belt/suicide_act(mob/living/carbon/user)
 	user.visible_message("<span class='suicide'>[user] begins belting themselves with \the [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
 	return BRUTELOSS
 
-/obj/item/storage/belt/update_icon()
-	cut_overlays()
-	if(content_overlays)
-		for(var/obj/item/I in contents)
-			var/mutable_appearance/M = I.get_belt_overlay()
+
+/obj/item/storage/belt/remove_from_storage(obj/item/I, atom/new_location)
+	. = ..()
+	if(. && is_type_in_typecache(I, content_overlays))

```

Comment:
For here and future reference, if the first parameter of `is_type_in_typecache` is for sure a datum, the faster way to do the check is `typecache[object.type]`